src/include/duckdb/function/table_function.hpp

```diff
@@ -468,6 +468,8 @@ class TableFunction : public SimpleNamedParameterFunction { // NOLINT: work-arou
 	bool late_materialization;
 	//! Additional function info, passed to the bind
 	shared_ptr<TableFunctionInfo> function_info;
+	//! Whether or not the in-out function is order dependent
+	bool in_out_function_order_dependent = false;
```

Comment:
I'm not sure why this needs to be tied exclusively to `in_out` functions - can we add:

```
OrderPreservationType order_preservation_type = OrderPreservationType::INSERTION_ORDER;
```

Which is used in `SourceOrder()` for regular table scans.